url5583f3dbb2_XOR19_XScript_tgz-pJ8-xscript_interactive_MainJ8/src/xscript/compiler/treemaker/XTreeMaker.java

```diff
@@ -0,0 +1,1486 @@
+			operators.add(o);
+			return new XTreeOperator(position, exprs, operators);
+		}
+	}
+	
+	private XTreeExpr makeExprPrefixSuffix(boolean inList){
+		startPosition();
+		startPosition();
+		XOperator operator;
+		List<XOperator> prefix = new ArrayList<XOperator>();
+		while(isOperator()){
+			operator = readOperator(Type.PREFIX, true);
+			if(operator==XOperator.NONE)
+				break;
+			prefix.add(0, operator);
+		}
+		if(prefix.isEmpty())
+			prefix = null;
+		XTreeExpr expr = makeExpr2(inList);
+		List<XOperator> suffix = new ArrayList<XOperator>();
+		while(isOperator()){
+			operator = readOperator(Type.SUFFIX, true);
+			if(operator==XOperator.NONE)
+				break;
+			suffix.add(operator);
+		}
+		if(suffix.isEmpty())
+			suffix = null;
+		if(suffix == null){
+			endPosition();
+		}else{
+			expr = new XTreeOperatorPrefixSuffix(endPosition(), null, expr, suffix);
+		}
+		XToken token = tokenizer.getToken();
+		while(token.kind==XTokenKind.KEYWORD && (token.data==XKeyword.LBRAKET || token.data==XKeyword.LINDEX || token.data==XKeyword.DOT || token.data==XKeyword.INSTANCEOF || token.data==XKeyword.ISSUBCLASS)){
+			startPosition();
+			startPosition();
+			if(token.data==XKeyword.LBRAKET){
+				expr = makeCallPosStarted(expr);
+			}else if(token.data==XKeyword.LINDEX){
+				tokenizer.next();
+				XTreeExpr index = makeExpr();
+				expect(XKeyword.RINDEX);
+				expr = new XTreeIndex(endPosition(), expr, index);
+			}else if(token.data==XKeyword.DOT){
+				tokenizer.next();
+				XTreeIdent ident = makeIdent();
+				List<XTreeExpr> exprs = new ArrayList<XTreeExpr>();
+				exprs.add(expr);
+				exprs.add(ident);
+				List<XOperator> operators = new ArrayList<XOperator>();
+				operators.add(XOperator.ELEMENT);
+				expr = new XTreeOperator(endPosition(), exprs, operators);
+			}else if(token.data==XKeyword.INSTANCEOF){
+				tokenizer.next();
+				XTreeExpr type = makeExprPrefixSuffix(inList);
+				expr = new XTreeInstanceof(endPosition(), expr, type);
+			}else if(token.data==XKeyword.ISSUBCLASS){
+				tokenizer.next();
+				XTreeExpr type = makeExprPrefixSuffix(inList);
+				expr = new XTreeIssubclass(endPosition(), expr, type);
+			}
+			suffix = new ArrayList<XOperator>();
+			while(isOperator()){
+				operator = readOperator(Type.SUFFIX, true);
+				if(operator==XOperator.NONE)
+					break;
+				suffix.add(operator);
+			}
+			if(suffix.isEmpty())
+				suffix = null;
+			if(suffix == null){
+				endPosition();
+			}else{
+				expr = new XTreeOperatorPrefixSuffix(endPosition(), null, expr, suffix);
+			}
+			token = tokenizer.getToken();
+		}
+		if(prefix == null){
+			endPosition();
+		}else if(expr instanceof XTreeOperatorPrefixSuffix){
+			expr.position.end = endPosition().end;
+			((XTreeOperatorPrefixSuffix) expr).prefix = prefix;
+		}else{
+			expr = new XTreeOperatorPrefixSuffix(endPosition(), prefix, expr, null);
+		}
+		return expr;
+	}
+	
+	private XTreeExpr makeExpr2(boolean inList){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 116 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.